tools/integration_tests/read_cache/small_cache_ttl_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package read_cache
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/client"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/log_parser/json_parser/read_logs"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/test_setup"
+)
+
+const (
+	objectName            = testDirName + "/" + testFileName
+	smallContent          = "small content"
+	smallContentSize      = 13
+	chunksReadAfterUpdate = 1
+	metadataCacheTTlInSec = 5
+)
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type smallCacheTTLTest struct {
+	flags         []string
+	storageClient *storage.Client
+	ctx           context.Context
+}
+
+func (s *smallCacheTTLTest) Setup(t *testing.T) {
+	mountGCSFuse(s.flags)
+	setup.SetMntDir(mountDir)
+	testDirPath = client.SetupTestDirectory(s.ctx, s.storageClient, testDirName)
+	client.SetupFileInTestDirectory(s.ctx, s.storageClient, testDirName, testFileName, fileSize, t)
+}
+
+func (s *smallCacheTTLTest) Teardown(t *testing.T) {
+	// unmount gcsfuse
+	setup.SetMntDir(rootDir)
+	unmountGCSFuseAndDeleteLogFile()
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test scenarios
+////////////////////////////////////////////////////////////////////////
+
+func (s *smallCacheTTLTest) TestSecondSequentialReadAfterUpdateIsCacheMiss(t *testing.T) {
+	// Read file 1st time.
+	expectedOutcome1 := readFileAndGetExpectedOutcome(testDirPath, testFileName, t)
+	validateFileInCacheDirectory(fileSize, s.ctx, s.storageClient, t)
+	client.ValidateObjectContentsFromGCS(s.ctx, s.storageClient, testDirName, testFileName,
+		expectedOutcome1.content, t)
+
+	// Append to the file.
+	err := client.WriteToObject(s.ctx, s.storageClient, objectName, smallContent, storage.Conditions{})
+	if err != nil {
+		t.Errorf("Could not append to file: %v", err)
+	}
+
+	// Read same file again immediately.
+	// Stale data would be served from cache in this case.
+	expectedOutcome2 := readFileAndGetExpectedOutcome(testDirPath, testFileName, t)
+	validateFileSizeInCacheDirectory(fileSize, t)
```

Comment:
If data is appended in the file, then how file size will be equal?

Reply:
in this case, stale data from cache is being served so that is why file size is same